lib/mesh.js

```diff
@@ -975,6 +980,63 @@ Mesh.prototype._initializeRootWebRoutes = function(callback) {
   callback();
 };
 
+Mesh.prototype.__getMiddlewareHandlers = function(middleware) {
+  const _this = this;
+  if (!middleware) throw new Error('undefined middleware in web config');
+  if (typeof middleware === 'function') return [middleware];
+  if (middleware.component == null) throw new Error(`web.middleware.component null in web config`);
+  if (middleware.methods == null || !middleware.methods.length)
+    throw new Error(
+      `web.middleware.methods empty or null in web config, component: ${middleware.component}`
+    );
+
+  const methods = middleware.methods.map(methodName => {
+    return function(req, res, next) {
+      if (!_this.exchange[middleware.component]) {
+        return next(new Error(`Component ${middleware.component} not found for global middleware`));
+      }
+      if (!_this.exchange[middleware.component][methodName]) {
+        return next(
+          new Error(`Method ${middleware.component}.${methodName} not found for global middleware`)
+        );
+      }
+      _this.exchange[middleware.component][methodName].call(
```

Comment:
is _this.exchange[middleware.component][methodName] not already bound to _this.exchange[middleware.component]?

Reply:
I have made it null, just did a quick test to ensure that the "this" was correct (which it was)